clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -166,155 +166,179 @@ void DeclRewriter::rewriteParmVarDecl(ParmVarDeclReplacement *N) {
 }
 
 
-template <typename DT, DeclReplacement::DRKind DK>
-void DeclRewriter::rewriteMultiDecl(DeclReplacementTempl<DT, DK> *N,
-                                    RSet &ToRewrite) {
-  DT *D = N->getDecl();
-  std::string SRewrite = N->getReplacement();
-  if (Verbose) {
-    errs() << "Decl at:\n";
-    if (N->getStatement())
-      N->getStatement()->dump();
-  }
-  SourceRange TR = D->getSourceRange();
+template <typename DRType>
+void DeclRewriter::rewriteFieldOrVarDecl(DRType *N, RSet &ToRewrite) {
+  static_assert(std::is_same<DRType, FieldDeclReplacement>::value
+                    || std::is_same<DRType, VarDeclReplacement>::value,
+                "Method expects variable or field declaration replacement.");
```

Comment:
I didn't know about `static_assert`. Are there other `assert`s in the code we could switch to be compile-time checkable, do you think?

Reply:
You're fairly limited on what can be checked in a static assert. It's mostly useful for adding stricter checks on template parameters. We don't use templates too heavily, but anywhere we do use them we might be able to add some compile time checks.